### PR TITLE
Implementing OpenAPI 3.1.0 spec

### PR DESCRIPTION
**Pull request by Nerzal**
**Describe the PR**
This PR is still work in progress. 

The PR makes use of this: spec library - https://github.com/sv-tools/openapi

What is done: 

1. Added a new CLI flag that tells the generator to use the new open api 3.1.0 specification
2. Added parsing of GeneralAPI Info

Ongoing: Parsing of operations

**Relation issue**
#548

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go: [ '1.16.x', '1.17.x', '1.18.x' ]
+        go: [ '1.20.x' ]
```
Comment (ubogdan) [inline .github/workflows/ci.yml:13]: Any special reason you are testing the code against 1.20 only ?

Comment (Nerzal) [reply .github/workflows/ci.yml:13]: The reason is the heavy use of generics. 
I could test if it also works with go 1.19

**Review by ubogdan — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Dockerfile References: https://docs.docker.com/engine/reference/builder/
 
 # Start from the latest golang base image
-FROM golang:1.18.3-alpine as builder
+FROM golang:alpine as builder
```
[the remark this answers is not in the document]
Comment (Nerzal) [reply Dockerfile]: fixed it. its not pinned to 1.19.7

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ubogdan) [reply .github/workflows/ci.yml]: Let's move the `spec.go` to its package so the package will work when it is imported into projects using an older version of go.

**Comment by ubogdan** [comment]
@Nerzal please let me know when the pr is done with the changes so we can merge it. Please mark the pr as a draft meanwhile.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Nerzal) [reply .github/workflows/ci.yml]: I just saw, that we can still support go 1.18 
image

**Review by ubogdan — APPROVED** [review]

**Comment by Nerzal** [comment]
I'm able to generate a complete documentation of our real world API. 
I'd say this is ready for a review and beta tests. 

Known Issues:

- Only application/json is supported as response data type right now. As it's hardcoded

**Comment by Nerzal** [comment]
@ubogdan can you tell my why the snyk test fails? 
I'm not allowed to see the reason

**Review by ubogdan — APPROVED** [review]

**Comment by ubogdan** [comment]
@Nerzal Thanks for your contribution!

**Comment by Nerzal** [comment]
Nice, I still have some more days for Bugfixing and writing more tests and implementing new features. 

So we should be able to get this to a production ready state pretty fast. 

And all thanks go to ClariLab for granting the time needed for this

**Comment by joy4eg** [comment]
@Nerzal  That's a really cool thing! Thank you so much for your effort!

**Comment by ubogdan** [comment]
@Nerzal published rc1 version

**Comment by darakanoit** [comment]
Are there any updates?
I have a problem with the fact that my other service only accepts open api 3.0 and I have to convert 2.0 to 3.0 myself before pushing

**Comment by ubogdan** [comment]
@darakanoit, this is an Open Source project. Please feel free to contribute, and we will be more than happy to assist you with CR to get this feature launched.

**Comment by Xantios** [comment]
What needs being done on this PR to get it merged? If there is anything i can do to contribute i'd love to make this happen.

**Pull request merged**